### PR TITLE
Add VersionedDocsLink and use for evergreen links to "latest" docs page

### LINKED ISSUE #174: "Learn more" link on front page is hard-coded to 2.19, should be latest stable
In #171 / https://github.com/pantsbuild/pants/issues/20648, we adjust the front page "getting started" link to not be dead, by hard-coding it to the current latest stable series: 2.19.

Once we release 2.20, this link should point to the `/2.20/...` page.

--- 

We have one page that computes a dynamic link using the latest version: https://github.com/pantsbuild/pantsbuild.org/blob/ac1d7051581bf4be77abd95b0405aa97202e6aad/src/pages/community/members.mdx#L28

```mdx
import { default as versions } from "@site/versions.json";
import Link from "@docusaurus/Link";

... <Link to={`/${versions[0]}/docs/contributions`}>helpful guide</Link>.
```
That links to the latest non-dev version (which could be an RC pre-release) not the latest stable version, so we may want to find a better way to do it (and wrap it up in a component? https://docusaurus.io/docs/markdown-features/react).

Preferably we retain link validity checking for it too (#66).

### PR DESCRIPTION
**Pull request by huonw**
This PR fixes #174 with a new `VersionedDocsLink` component that reads the docusaurus config to determine what the latest stable version is.

It also adjusts the other dynamic link to the contribution guide to use this. This link is changed from the latest RC versions (currently 2.20) to the current `main` version (2.21), i.e. the absolute most up-to-date version of the contribution guide, which seems most relevant to contributors since they'll be starting in the `main` branch.

This has a few issues with the component due to https://github.com/prettier/prettier/issues/12209, which forces us into a very long line on the `_index.mdx` page.

I'm unclear if a broken link will be detected or not.

### CODE REVIEW

File: src/pages/_index.mdx [hunk 1]
```diff
@@ -116,7 +116,10 @@ You'll find no subsets like Starlark here. Pants empowers you with full support
 
 Pants supports Python, Docker, Go, Java, Kotlin, Pex, Protodoc, Scala, Shell, Thrift, Protobuf,
 Docker, Helm, many linting and formatting tools, packaging, coverage, and more.
-[Learn more.](/2.19/docs/introduction/welcome-to-pants)
+
+<VersionedDocsLink version="last-stable" unversionedPath="docs/broken-link">
+  Learn more.
+</VersionedDocsLink>
```
Comment (huonw) [inline src/pages/_index.mdx:122]: Note to self: this spacing change is forced by prettier but is changing the display.

Comment (huonw) [reply src/pages/_index.mdx]: https://github.com/prettier/prettier/issues/12209

File: src/pages/community/members.mdx [hunk 2]
```diff
@@ -25,7 +24,7 @@ Contributions come in many forms, and we appreciate all of them! Examples includ
 
 Whatever your area of expertise and your skill level, there may be valuable contributions you can make. Are you a graphic designer? A technical writer? Do you know how to make videos? There might be a cool contribution in your future.
 
-We try and make contributions easy. For example, you can suggest documentation fixes by clicking on the Suggest Edits link on any page. And you can report bugs by opening a [GitHub issue](https://github.com/pantsbuild/pants/issues). If you want to hack on the Pants codebase itself there is a <Link to={`/${versions[0]}/docs/contributions`}>helpful guide</Link>.
+We try and make contributions easy. For example, you can suggest documentation fixes by clicking on the Suggest Edits link on any page. And you can report bugs by opening a [GitHub issue](https://github.com/pantsbuild/pants/issues). If you want to hack on the Pants codebase itself there is a <VersionedDocsLink version="current-dev" unversionedPath="docs/contributions">helpful guide</VersionedDocsLink>.
```
Comment (huonw) [inline src/pages/community/members.mdx:27]: As noted in the PR description this is changing from the first entry of `versions.json` to the "current". This means switching from the most recent `versioned_docs/...` entry (which corresponds to some `2.*.x` branch in https://github.com/pantsbuild/pants) to the contents of `docs/` (which corresponds to `main`).

I think this is reasonable: people are generally contributing to the `main` branch, so the docs there are what's relevant.

(For this particular one, arguably the "process" parts of the docs like https://www.pantsbuild.org/2.20/docs/contributions are "evergreen" pages and could be moved to this repo, while some pages like https://www.pantsbuild.org/2.20/docs/contributions/development/developing-rust are more strongly tied to a particular version and so aren't evergreen.)

**Review by thejcannon — APPROVED** [review]
Very neat!

**Pull request merged**